doc/relevancy.md

```diff
@@ -0,0 +1,442 @@
+to match. With a high enough slop value, words can be arranged in any order.
+
+https://www.elastic.co/guide/en/elasticsearch/guide/current/slop.html
+
+Shingles are more lenient since we're only looking for pairs of words instead
+of requiring the whole phrase to match. **But our implementation is broken.**
+This is one of the things that was fixed in the new search weighting.
+
+See Richard's commit message: https://github.com/alphagov/search-api/commit/6ece793190932fee31be5437c4c56773dc358ff6
+
+>  The existing weighting only applies this analysis at search time, so the
+>  pairs it looks for never exist in the index. This commit instead
+>  creates a new pseudo-subfield for searchable text fields which is
+>  indexed and searched with shingles applied. The weights from searches
+>  across this field are used to boost scores of documents.
+
+### There is no boosting of individual field matches
+
+We treat a match in title, description, acronym and indexable content the same,
+but these are completely different things.
+I would expect title and description matches to be much more important than
+indexable content.
+
+For some reason we're already using field-specific boosts when the query
+is a quoted phrase, but not for normal searches.
+
+```
+PHRASE_MATCH_TITLE_BOOST = 5
+PHRASE_MATCH_ACRONYM_BOOST = 5
+PHRASE_MATCH_DESCRIPTION_BOOST = 2
+PHRASE_MATCH_INDEXABLE_CONTENT_BOOST = 1
+```
+
+### We have lots of code stuck behind feature flags
+
+These things are behind a feature flag and are not actually used in production:
+
+- ID code handling
+- All the new weighting stuff, including improvements to synonyms and shingles
+
+### Popularity is the main proxy for "good"
+
+Popularity is one of the biggest factors influencing the weighting. It's
+based on the overall page views, so there's a positive feedback loop.
+`Thing gets popular -> thing becomes more visible -> thing gets more popular`
+
+In the past this has caused problems when we know content is better but
+something else still gets high traffic.
+https://github.com/alphagov/search-api/commit/03c4bfa0a1a816a57d38b71ac5cb22c3a107c275
+
+If we had other signals of content quality we might be able to reduce our
+dependence on the page views.
+
+### Recommendations
+
+- Replace shingles implementation with one that works
+- A/B test with/without `match_phrase` component
+- Boost title/description matches in the match_all_terms, and match_bigrams subqueries
+- Remove unused code behind feature flags by end of quarter
+
+
+## Analyzer usage
+
+This table describes what each of the analyzers do:
+
+| Analyzer                | Used by main query | Details                                                                 | Normalize quotes | Strip Quotes | Tokenize                 | Trim | Lowercase | Asciifolding | Old school synonyms | Remove stop words | Synonyms            | Mark protwords (not stemmed) | Stemmer override | Porter2 stemmer | Shingles      | ID codes analysis |
```

Comment:
Should link to the code at config/schema/elasticsearch_schema.yml, as this won't get out of date.

Should also briefly explain what an analyzer is and link to https://www.elastic.co/guide/en/elasticsearch/reference/current/analyzer.html.

An index has various analyzers. At index time, elasticsearch will use an analyzer named default in the index settings if no analyzer is specified in the field mapping: https://www.elastic.co/guide/en/elasticsearch/reference/current/analyzer.html (will this still be true in version 6?).

We don't need this at query time unless synonyms are disabled using a debug flag.

Reply:
I've added a detail on what an analyzer is, and a link to the docs.